platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -1746,22 +1746,32 @@ public boolean createTransaction(final byte[] trans) {
     @Override
     public PlatformEvent[] getAllEvents() {
         final EventImpl[] allEvents = shadowGraph.getAllEvents();
-        Arrays.sort(allEvents, (o1, o2) -> {
-            if (o1.getConsensusOrder() != -1 && o2.getConsensusOrder() != -1) {
-                // both are consensus
-                return Long.compare(o1.getConsensusOrder(), o2.getConsensusOrder());
-            } else if (o1.getConsensusTimestamp() == null && o2.getConsensusTimestamp() == null) {
-                // neither are consensus
-                return o1.getTimeReceived().compareTo(o2.getTimeReceived());
-            } else {
-                // one is consensus, the other is not
-                if (o1.getConsensusTimestamp() == null) {
-                    return 1;
-                } else {
-                    return -1;
-                }
+        // There is currently a race condition that can cause an exception if event order changes at
+        // just the right moment. Since this is just a testing utility method and not used in production
+        // environments, we can just retry until we succeed.
+        while (true) {
```

Comment:
This feels like a code smell.   If the sorting throws an exception and we are constantly restarting on sorting, what guarantees termination of the method? Is there any way to detect that the properties in the order are settled and won't change during execution of the sort, and can we wait for that condition to become true?

Reply:
Added a maximum number of retries before we crash.

The reason why this is crashing is because of a known thread safety issue.  The `Arrays.sort()` method requires that the result of `EventImpl.compareTo()` not change between the start of the call and the end of the call. This is a logical requirement, since you can't sort something if the ordering keeps changing mid-sort. However, we are actively changing event order as we receive more events in the shadowgraph.

This method is deprecated, and the only remaining use case is when you are running the local demo UI. We don't consider this to be "production code". Because of that reason, it's hard to justify the time and effort required to properly fix this (at least in the short term). My change is a band-aid that prevent the local demo from crashing (the node will crash if this is uncaught).